Add api.neuvector.com api group resources to role creation

jordojordo opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #11174 
<!-- Define findings related to the feature or bug issue. -->

This adds the resources for the `api.neuvector.com` api group to the Global, Cluster, and Project/Namespace role editing pages.

### Occurred changes and/or fixed issues
<!-- Include information of the changes, including collateral areas which have been affected by this PR as requirement or for convenience. -->
The resources for the apiGroups is a hardcoded list for each role type, this adds the resources for the neuvector group.

### Technical notes summary
<!-- Outline technical changes which may pass unobserved or may help to understand the process of solving the issue -->

### Areas or cases that should be tested
<!-- Areas that should be tested can include Airgap checks, Rancher upgrades, K8s upgrade, etc. -->
<!-- Which browser did you use for local testing? The reviewer should test with a different browser. -->
<!-- Add missing steps or rewrite them if have been missed or to complement existing information. This should define a clear way to reproduce it and not an approximation. -->
Creating a Global, Cluster, and Project/Namespace role with any of the neuvector resources.

### Areas which could experience regressions
<!-- Create a detailed list of areas to be analyzed which may be affected by the changes, which would require a prior research to avoid regressions. -->

### Screenshot/Video
<!-- Attach screenshot or video of the changes and eventual comparison if you find it necessary -->

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

nwmac reviewed (changes requested):
How did you test this?

For cluster roles, the resource selector freezes for me and I see lots of these in the browser console:

![image](https://github.com/rancher/dashboard/assets/1955897/1c9a0391-62b3-41a9-b4d8-4ba9be5d07ee)

Looking at the code (and a quick test), I don't think you need to add the resources to both global and cluster - just adding to global works - looks like we use both for cluster, so with them in both places, you get the error above.

 Could you expand on the manual test details and maybe add an example screen shot.

jordojordo commented:
I needed to create unique labels for both the cluster scoped and namespaced roles, in the RoleDetailEdit component I added a condition to check for the `neuvectorApi` group and apply unique labels based on the scope.

Also updated the manual testing steps.
